position_in_partition.hh

```diff
@@ -599,22 +599,25 @@ public:
 };
 
 struct view_and_holder {
-    position_in_partition_view view;
-    std::optional<position_in_partition> holder;
+    std::variant<position_in_partition_view, position_in_partition> pos;
```

Comment:
Couldn't we just define a move constructor which recreates the view after move?
This variant is so ugly.

Reply:
We could, but this version, even though it has a variant, is more straightforward. What do you dislike about the variant?

Reply:
`view()` is now a call to `visit()` which is pretty inefficient I hear. Maybe not a problem here.

Reply:
I think a move constructor is better. I don't know how critical the code here, but it's low-level enough to avoid slowing it down unnecessarily.

variant in libstdc++ is coded badly, it generates indirect calls through a table of function pointers instead of direct code that can be optimized away.

Reply:
Ok, I'll take the move constructor approach then